hail/python/hailtop/aiotools/weighted_semaphore.py

```diff
@@ -23,16 +24,18 @@ class WeightedSemaphore:
     def __init__(self, value: int):
         self.max = value
         self.value = value
-        self.events: List[Tuple[int, asyncio.Event]] = []
+        self.events: SortedKeyList[Tuple[int, asyncio.Event]] = SortedKeyList(key=lambda x: x[0])
```

Comment:
Why did you choose a SortedKeyList rather than a SortedSet? I'm not sure if there's a huge performance tradeoff between the two.

Reply:
asyncio.Event doesn't have comparison defined so SortedList fails when there are two tuples with the same weight.